typescript/sdk/src/providers/transactions/submitter/ethersV5/types.ts

```diff
@@ -39,3 +40,43 @@ export const EV5ImpersonatedAccountTxSubmitterPropsSchema =
 export type EV5ImpersonatedAccountTxSubmitterProps = z.infer<
   typeof EV5ImpersonatedAccountTxSubmitterPropsSchema
 >;
+export type EvmIcaTxSubmitterProps = z.infer<
+  typeof EvmIcaTxSubmitterPropsSchema
+>;
+
+export const EvmIcaTxSubmitterInternalSubmitterConfigSchema = z
+  .discriminatedUnion('type', [
+    z.object({
+      type: z.literal(TxSubmitterType.JSON_RPC),
+    }),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.GNOSIS_TX_BUILDER),
+      })
+      .merge(EV5GnosisSafeTxBuilderPropsSchema.omit({ chain: true })),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.GNOSIS_SAFE),
+      })
+      .merge(EV5GnosisSafeTxSubmitterPropsSchema.omit({ chain: true })),
+    z
+      .object({
+        type: z.literal(TxSubmitterType.IMPERSONATED_ACCOUNT),
+      })
+      .merge(
+        EV5ImpersonatedAccountTxSubmitterPropsSchema.omit({ chain: true }),
+      ),
+  ])
+  .default({
+    type: TxSubmitterType.JSON_RPC,
+  });
```

Comment:
instead of duplicating the union here, can we just make this a "recursive" definition and have internalSubmitter be the top level submitter schema?

Reply:
otherwise this will be come stale when new submitters are added

Reply:
Oky, should we then add a limit on the nesting it can be done when creating the strategy config? I created the subtype to make it more difficult to create an invalid/overly complex config